src/js/visual/voicecontrols.js

```diff
@@ -31,13 +31,17 @@ function createVoiceControls(audioQueue) {
     // Set voice and rate to match initial form values
     audioQueue.setRate(parseFloat(rateValue.textContent));
     audioQueue.setVoice(voiceSelect.value);
-  })
+  });
 
-  // Update voices when they change
-  window.speechSynthesis.onvoiceschanged = function () {
-    voiceSelect.innerHTML = ""; // Clear existing options
-    populateVoices();
-  };
+  // Android Webview doesn have speechSynthesis 
```

Comment:
The web page works in Chrome on Android, right? I think what you're saying here is that the Capacitor-based app uses an Android Webview which doesn't implement the Web Speech API, but presumably that's where the Capacitor plugin interfaces with the native OS services?

Reply:
Yes, what Capacitor does is run the web app inside a WebView within a real app. It provides access to native components for JavaScript using different plugins, enabling interaction with native components from different platforms.